src/proxyvisor/utils.coffee

```diff
@@ -0,0 +1,78 @@
+Promise = require 'bluebird'
+_ = require 'lodash'
+fs = Promise.promisifyAll require 'fs'
+path = require 'path'
+mkdirp = Promise.promisify(require('mkdirp'))
+execAsync = Promise.promisify(require('child_process').exec)
+
+self = module.exports =
+
+	isDefined: _.negate(_.isUndefined)
+
+	parseDeviceFields: (device) ->
+		device.appId = parseInt(device.appId)
+		device.environment = JSON.parse(device.environment ? '{}')
+		device.targetEnvironment = JSON.parse(device.targetEnvironment ? '{}')
+		device.config = JSON.parse(device.config ? '{}')
+		device.targetConfig = JSON.parse(device.targetConfig ? '{}')
+		return _.omit(device, 'deviceId', 'markedForDeletion', 'logs_channel')
+
+	parseApplicationFields: (application) ->
+		application.appId = parseInt(application.appId)
+		application.parentApp = parseInt(application.parentApp)
+		application.environment = JSON.parse(application.environment ? '{}')
+		application.config = JSON.parse(application.config ? '{}')
+		return _.omit(application, 'id', 'releaseId', 'image')
+
+	# TODO move to lib/validation
+	validStringOrUndefined: (s) ->
+		_.isUndefined(s) or !_.isEmpty(s)
+	validObjectOrUndefined: (o) ->
+		_.isUndefined(o) or _.isObject(o)
+
+	tarDirectory: (appId) ->
+		return "/data/dependent-assets/#{appId}"
+
+	tarFilename: (appId, commit) ->
+		return "#{appId}-#{commit}.tar"
+
+	tarPath: (appId, commit) ->
+		return "#{self.tarDirectory(appId)}/#{self.tarFilename(appId, commit)}"
+
+	getTarArchive: (source, destination) ->
+		fs.lstatAsync(destination)
+		.catch ->
+			mkdirp(path.dirname(destination))
+			.then ->
+				execAsync("tar -cvf '#{destination}' *", cwd: source)
+
+	cleanupTars: (appId, commit) ->
```

Comment:
Check the latest version of this function - there's been changes to this that are getting lost in the rebase